libc/dirent/rewinddir.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * test/test_strftime.c
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Arkadiusz Kozlowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <time.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <dirent.h>
+#include <limits.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <string.h>
+
+#include <unity_fixture.h>
+
+#include "common.h"
+
+#define MAIN_DIR "test_rewinddir"
+
+
+TEST_GROUP(dirent_rewinddir);
```

Comment:
You should also make testcase:

- Not only direcotry, but also files, symbolic links and other types of files
- It shall also cause the directory stream to refer to the current state of the corresponding directory, as a call to `opendir()` would have done. - this is not tested
- After a call to the [fork()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/fork.html) function, either the parent or child (but not both) may continue processing the directory stream using [readdir()](https://pubs.opengroup.org/onlinepubs/9699919799/functions/readdir.html), rewinddir(). If both the parent and child processes use these functions, the result is undefined. - this also not tested